File: app/src/main/java/com/afkanerd/deku/QueueListener/GatewayClients/GatewayClientProjectListingViewModel.java
package com.afkanerd.deku.QueueListener.GatewayClients;

import android.content.Context;
import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import androidx.room.Room;

import com.afkanerd.deku.DefaultSMS.Models.Database.Datastore;
import com.afkanerd.deku.DefaultSMS.Models.ThreadingPoolExecutor;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class GatewayClientProjectListingViewModel extends ViewModel {

    long id;
    MutableLiveData<List<GatewayClientProjects>> mutableLiveData = new MutableLiveData<>();
    public LiveData<List<GatewayClientProjects>> get(Datastore databaseConnector, long id) {
        this.id = id;
        GatewayClientProjectDao gatewayClientProjectDao = databaseConnector.gatewayClientProjectDao();
        ThreadingPoolExecutor.executorService.execute(new Runnable() {
            @Override
            public void run() {
                mutableLiveData.postValue(gatewayClientProjectDao.fetchGatewayClientIdList(id));
            }
        });
        return mutableLiveData;
    }

}

File: app/src/main/java/com/afkanerd/deku/QueueListener/GatewayClients/GatewayClientProjectListingViewModel.kt
package com.afkanerd.deku.QueueListener.GatewayClients

import android.content.Context
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.afkanerd.deku.Datastore
import com.afkanerd.deku.Modules.ThreadingPoolExecutor

class GatewayClientProjectListingViewModel : ViewModel() {
    private lateinit var datastore: Datastore

    private lateinit var liveData : LiveData<List<GatewayClientProjects>>
    fun get(context: Context, gatewayClientId: Long): LiveData<List<GatewayClientProjects>>{
        datastore = Datastore.getDatastore(context)
        if(!::liveData.isInitialized) {
            liveData = MutableLiveData()
            liveData = datastore.gatewayClientProjectDao().fetchGatewayClientId(gatewayClientId)
        }
        return liveData
    }

    fun insert(gatewayClientProjects: GatewayClientProjects) {
        datastore.gatewayClientProjectDao().insert(gatewayClientProjects)
    }

    fun update(gatewayClientProjects: GatewayClientProjects) {
        datastore.gatewayClientProjectDao().update(gatewayClientProjects)
    }
}
